src/commands/new.ts

```diff
@@ -81,17 +81,20 @@ export default class New extends Command {
   async createAsyncapiFile(fileName:string) {
     const defaultAsyncapiFile = await readFile(resolve(__dirname, '../../assets/asyncapi.yaml'), { encoding: 'utf8' });
 
+    const fileNameHasFileExtension = fileName.split('.').pop() !== fileName;
```

Comment:
What about this?

```suggestion
    const fileNameHasFileExtension = fileName.includes('.')
```

Creating an array and removing elements is an expensive operation. I think we can just check for the presence of a `.` char.

Reply:
Yeah happy to go with that, would only mean things like `application.random.yaml` would not be valid, although I have no idea who would do this stuff.... pop means we always get the last one regardless of the amount of `.` in the filename.

What do you think? Am I over thinking it :D?

Reply:
But if someone does `application.random.yaml` then it means `application.random` is the name, isn't it?

Reply:
None of our solutions will cover that case, I think. Unless we obtain the extension and check if it's `yml`, `yaml`, or `json` and therefore dicard `random` or any other string 🤔

Reply:
> None of our solutions will cover that case, I think. Unless we obtain the extension and check if it's yml, yaml, or json and therefore dicard random or any other string 🤔

Yeah think you are right. Hmmm think let's go for includes and see if we get any issues about it, and in future if we do we can sniff out the extensions we want